presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1727,9 +1731,14 @@ private void verifyJvmTimeZone()
     private static HiveStorageFormat extractHiveStorageFormat(Table table)
     {
         StorageFormat storageFormat = table.getStorage().getStorageFormat();
-        String outputFormat = storageFormat.getOutputFormat();
+        String outputFormat;
+        try {
+            outputFormat = storageFormat.getOutputFormat();
+        }
+        catch (IllegalStateException e) {
```

Comment:
@kokosing  thanks for your comment;
```Why not to throw the proper exception from storageFormat.getOutputFormat() directly?```
I think this will break the code coherence because from the `StorageFormat` layer, it just know that the `storageFormat` null and then throws `IllegalStateException`, it didn't care anything about Presto. You could see that any exception in `StorageFormat` are `IllegalStateException`; So, I think it is better to remain the same in `StorageFormat`. But when it comes to the caller, the method `extractHiveStorageFormat()`, it is responsible to process this weird-looking  `IllegalStateException`, then it comes be `PrestoException`;

-----------
```Or even better why create incorrectStorageFormat in a first place and not to throw such exception in the constructor of StorageFormat?```

I think it is reasonable. You could see that `StorageFormat` didn't do any check for storage format in method: 
```
    public static StorageFormat fromHiveStorageFormat(HiveStorageFormat hiveStorageFormat)
    {
        return new StorageFormat(hiveStorageFormat.getSerDe(), hiveStorageFormat.getInputFormat(), hiveStorageFormat.getOutputFormat());
    }

  private StorageFormat(String serDe, String inputFormat, String outputFormat)
    {
        this.serDe = serDe;
        this.inputFormat = inputFormat;
        this.outputFormat = outputFormat;
    }

```
Do you think whether or not it is just expected?

Reply:
Right, it is a bit more complicated. All the fields are nullable. It looks like you could either: 
 - add a method to `StorageFormat` like is `StorageFormat#isNullable` and then you check it instead of catching exception.
 - or (better a bigger change) you could make `StorageFormat` to return `Optional` values.

What do you think?

Reply:
Looking at where this is used, we would also get this `IllegalStateException` if we try to insert into a table with a null storage format. Also, `getSerDe()` has the same problem. I think it's better if we change the `StorageFormat` methods like this:
```java
if (serDe == null) {
    throw new PrestoException(HIVE_INVALID_METADATA, "SerDe is not present in StorageFormat");
}

Reply:
@kokosing  
It seems that when we make some check like `requireNonNull` when creating StorageFormat, we are just making this exception occur earlier, but it in fact didn't solve the problem that one single table null serde failed the whole table-scan query; am I right?

Reply:
> I think it's better if we change the `StorageFormat` methods like this:

You think we'd better change the IllegalStateException to PrestoException in StorageFormat method like `getSerDe` and `getInputFormat` and `getOutputFormat` ?

Reply:
@electrum  Do you have any comments ?  Let's move it forward.